Craft changes

ericmeek opened the pull request:
Workorders for all disciplines work with hometown setting now (Crossing & Riverhaven)
Parts has it's own section (recipe_parts) in base-crafting.yaml
All room #'s taken out of crafting scripts and pulled from base-crafting.yaml
Carve script updated with get_item/stow_item same as shape script
ensure_copper_on_hand in common-money updated to check hometown

rpherbig commented on common-crafting.lic:
```diff
@@ -12,26 +12,26 @@ module DRCC
   end
 
-  def find_empty_crucible
+  def find_empty_crucible(idle_room = nil, crucibles)
```
Would it be any easier if you passed in a hometown string as an argument and had these helpers do the lookup?

ericmeek replied:
Yes, I should have the lookups in the common- files. Changing that.

On Tue, Sep 6, 2016 at 12:41 PM, Robert Herbig <notifications@github.com>
wrote:

> In common-crafting.lic
> <https://github.com/rpherbig/dr-scripts/pull/751#discussion_r77672027>:
>
> > @@ -12,26 +12,26 @@ module DRCC
> >      ['nothing in there'].include?(DRC.bput('look in cruc', 'nothing in there', 'you see'))
> >    end
> >
> > -  def find_empty_crucible
> > +  def find_empty_crucible(idle_room = nil, crucibles)
>
> Would it be any easier if you passed in a hometown string as an argument
> and had these helpers do the lookup?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/rpherbig/dr-scripts/pull/751/files/d56664e67f7772e62511263960f200f4536e4317#r77672027>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ARQKjsB6cksjymBOOAHd6bkF5WtGSi43ks5qnZeegaJpZM4J2CdL>
> .
>

ericmeek replied:
Actually I think I'll have the default argument lookup the hometown and it
can be overridden if in some case someone wanted to. Although I can't think
of a reason anyone would. See any issues with that?

On Tue, Sep 6, 2016 at 1:42 PM, Eric Meek <eric.meek71@gmail.com> wrote:

> Yes, I should have the lookups in the common- files. Changing that.
>
> On Tue, Sep 6, 2016 at 12:41 PM, Robert Herbig <notifications@github.com>
> wrote:
>
>> In common-crafting.lic
>> <https://github.com/rpherbig/dr-scripts/pull/751#discussion_r77672027>:
>>
>> > @@ -12,26 +12,26 @@ module DRCC
>> >      ['nothing in there'].include?(DRC.bput('look in cruc', 'nothing in there', 'you see'))
>> >    end
>> >
>> > -  def find_empty_crucible
>> > +  def find_empty_crucible(idle_room = nil, crucibles)
>>
>> Would it be any easier if you passed in a hometown string as an argument
>> and had these helpers do the lookup?
>>
>> —
>> You are receiving this because you authored the thread.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/rpherbig/dr-scripts/pull/751/files/d56664e67f7772e62511263960f200f4536e4317#r77672027>,
>> or mute the thread
>> <https://github.com/notifications/unsubscribe-auth/ARQKjsB6cksjymBOOAHd6bkF5WtGSi43ks5qnZeegaJpZM4J2CdL>
>> .
>>
>
>

rpherbig replied:
I'm not sure what you mean. Something like `def foo(hometown='Crossing')`?

ericmeek replied:
This:

def foo(hometown = get_settings["hometown"])

Could be done a couple different ways as long as it can be called with
defaults or be overridden.

On Tue, Sep 6, 2016 at 1:57 PM, Robert Herbig <notifications@github.com>
wrote:

> In common-crafting.lic
> <https://github.com/rpherbig/dr-scripts/pull/751#discussion_r77686257>:
>
> > @@ -12,26 +12,26 @@ module DRCC
> >      ['nothing in there'].include?(DRC.bput('look in cruc', 'nothing in there', 'you see'))
> >    end
> >
> > -  def find_empty_crucible
> > +  def find_empty_crucible(idle_room = nil, crucibles)
>
> I'm not sure what you mean. Something like def foo(hometown='Crossing')?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/rpherbig/dr-scripts/pull/751/files/d56664e67f7772e62511263960f200f4536e4317#r77686257>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ARQKjuewAUd9anoX3RUmfa8lNLXhUy0Sks5qnamZgaJpZM4J2CdL>
> .
>

rpherbig replied:
I'm not a big fan of calling get_settings from the helpers since it won't account for args that were passed to the script which load other setting files.

ericmeek replied:
Ahh, didn't realize that. Hometown string it is.

On Tue, Sep 6, 2016 at 2:22 PM, Robert Herbig <notifications@github.com>
wrote:

> In common-crafting.lic
> <https://github.com/rpherbig/dr-scripts/pull/751#discussion_r77690814>:
>
> > @@ -12,26 +12,26 @@ module DRCC
> >      ['nothing in there'].include?(DRC.bput('look in cruc', 'nothing in there', 'you see'))
> >    end
> >
> > -  def find_empty_crucible
> > +  def find_empty_crucible(idle_room = nil, crucibles)
>
> I'm not a big fan of calling get_settings from the helpers since it won't
> account for args that were passed to the script which load other setting
> files.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/rpherbig/dr-scripts/pull/751/files/d56664e67f7772e62511263960f200f4536e4317#r77690814>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ARQKjnhOy8FjLCdo4vhuWz4MUWxaGV-Qks5qna93gaJpZM4J2CdL>
> .
>

ericmeek commented:
Helpers in common-crafting.lic now take hometown string as argument
Updated ensure_copper_on_hand method
Updated workorders.lic and individual crafting scripts re: changes in common-crafting
Tested workorders for all disciplines in Crossing & Haven

rpherbig commented:
It looks like this cannot be automatically merged due to conflicts. Can you merge master back into your branch and fix the conflicts?

rpherbig commented on forge.lic:
```diff
@@ -91,7 +92,12 @@ class Forge
     stow_item('book')
-    find_grindstone
+    
+    settings = get_settings
```
Are lines 96, 97, 98 still necessary?

See also lines 111, 112, 113, 307, 308, 309.

rpherbig commented on makesteel.lic:
```diff
@@ -1,13 +1,17 @@
 count = variable.last.to_i
 
-DRCM.ensure_copper_on_hand(2000 * count)
+get_settings["hometown"]
```
Can this line be removed?

rpherbig commented on workorders.lic:
```diff
@@ -250,7 +261,9 @@ class WorkOrders
     order_yarn(info['stock-room'], stock_needed, info['stock-number'])
 
-    find_sewing_room
+    data = get_data('crafting')['tailoring']['Riverhaven']
```
Why is Riverhaven hardcoded here?

rpherbig commented on workorders.lic:
```diff
@@ -250,7 +261,9 @@ class WorkOrders
-    find_sewing_room
+    data = get_data('crafting')['tailoring']['Riverhaven']
+    find_sewing_room(@hometown)
+    #find_sewing_room(@engineering_room)
```
Can this line be removed?

rpherbig commented on common-money.lic:
```diff
@@ -33,16 +33,18 @@ module DRCM
+  def wealth(hometown)
+    case hometown
+    when "Riverhaven"
+      wealth = DRC.bput("wealth lirum", '\(\d+ copper Lirums\)', "No Lirums").scan(/\d+/).first.to_i
```
You've got a map between hometown and its currency. That data is already in `base-town` (hometown['exchange']['currency']).

I think we should move currency up a level so it is a property of the town itself, but that's not necessary in this PR.
